src/Tribe/Template.php

```diff
@@ -502,13 +502,27 @@ public function get_template_file( $name ) {
 	 * @return string|false Either the final content HTML or `false` if no template could be found.
 	 */
 	public function template( $name, $context = array(), $echo = true ) {
-		// If name is String make it an Array
-		if ( is_string( $name ) ) {
-			$name = (array) explode( '/', $name );
+		static $file_exists    = [];
+		static $files          = [];
+		static $template_names = [];
+
+		// Key we'll use for in-memory caching of expensive operations.
+		$cache_name_key = $name;
+
+		// Cache template name massaging so we don't have to repeat these actions.
+		if ( ! isset( $template_names[ $cache_name_key ] ) ) {
+			// If name is String make it an Array
+			if ( is_string( $name ) ) {
+				$name = (array) explode( '/', $name );
+			}
+
+			// Clean this Variable
```

Comment:
:no_entry_sign: Inline comments must end in full-stops, exclamation marks, or question marks

<sub>hash: ae5c50d64c2a836532d7d25085188cb9</sub>